lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +209,28 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ''
+        for div in article_bs.find_all('div', class_='news_item_name'):
+            for a in div.find_all('a'):
+                url = self.url_pattern + a['href']
+                if url not in self.urls:
+                    return url
+        return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        while len(self.urls) < self.config.get_num_articles():
+            for seed_url in self.get_search_urls():
+                response = make_request(seed_url, self.config)
+                if not response.ok:
+                    continue
+                soup = BeautifulSoup(response.text, 'lxml')
+                extracted_urls = self._extract_url(soup)
+                self.urls.append(extracted_urls)
+                if len(self.urls) >= self.config.get_num_articles():
+                    break
```

Comment:
Do you really need this breakage considering you use `while` cycle?

Reply:
not really, thank you